test/unit/resources/etc_hosts_allow_deny_test.rb

```diff
@@ -23,6 +23,32 @@
       _(entries.client_list).must_include ['127.0.1.154',  '[:fff:fAb0::]']
     end
   end
+
+  describe '#parse_line' do
```

Comment:
This isn't quite what I was suggesting, though it's fairly effective.

You can test individual methods, even if they are private (and we should make sure `parse_line` is private, please!). For example:

```ruby
it 'parses a line with multiple clients' do
  line = 'foo: client1, client2 : some_option'
  _(resource.send(:parse_line, line)).must_equal [ 'client1', 'client2 ' ]
end
```

This allows you test lots and lots of iterations of that method accepting different values by just focusing on that method - you don't exercise FilterTable, the mocked data, etc. Your test remains laser-focused. Given the importance of this one method, we should isolate a number of tests just on that method to ensure it works properly and catch any future regressions.

Reply:
Okay sounds good, Only test you suggested that wouldnt make sense is to test a line with multiple daemons because I split up lines with mulitiple daemons before that method is ever called.